### ISSUE TITLE
sharding-jdbc: Oracle pagination query error

### ISSUE THREAD

**Issue by pcyan** [apache/shardingsphere#17477]
version
```
<groupId>org.apache.shardingsphere</groupId>
<artifactId>shardingsphere-jdbc-core-spring-boot-starter</artifactId>
<version>5.1.1</version>
```

testcase
```java
def "testOraclePagination"() {

        when: "pagination case 1"
        def pagination_sql2 = """
    select * from (
    select t1."num",rownum as "rownum_" from(
        select 'a'  as "num" from dual)t1)
    where "rownum_" >? and "rownum_"< ?
"""

        try (def connection = sqlSessionFactory.openSession().getConnection()) {
            connection.prepareStatement(pagination_sql2)
        }
        then:
        noExceptionThrown()

        when: "pagination case 2"
        def pagination_sql3 = """
    select t2."num",t2,"rownum_" from (
    select t1."num",rownum as "rownum_" from(
        select 'a'  as "num" from dual)t1)t2
    where t2."rownum_" > ? and t2."rownum_" <= (?+?)
"""

        try (def connection = sqlSessionFactory.openSession().getConnection()) {

            connection.prepareStatement(pagination_sql3)
        }
        then:
        noExceptionThrown()


        when: "pagination case 3"
        def pagination_sql4 = """
    select * from (
    select t1."num",rownum as "rownum_" from(
        select 'a'  as "num" from dual)t1)
    where t2."rownum_" > ? and t2."rownum_" <= (?+?)
"""

        try (def connection = sqlSessionFactory.openSession().getConnection()) {

            connection.prepareStatement(pagination_sql4)
        }
        then:
        noExceptionThrown()
    }
```

pagination case 3 throw exception
```
Caused by: java.lang.ClassCastException: class org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.complex.CommonExpressionSegment cannot be cast to class org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.simple.ParameterMarkerExpressionSegment (org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.complex.CommonExpressionSegment and org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.simple.ParameterMarkerExpressionSegment are in unnamed module of loader 'app')
	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.RowNumberPaginationContextEngine.createRowNumberValueSegment(RowNumberPaginationContextEngine.java:142)
	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.RowNumberPaginationContextEngine.createPaginationWithRowNumber(RowNumberPaginationContextEngine.java:129)
	at org.apache.shardingsphere.infra.binder.segment.select.pagination.engine.RowNumberPaginationContextEngine.createPaginationContext(RowNumberPaginationContextEngine.java:67)

```

**Comment by strongduanmu** [comment]
Hi @pcyan, thank you for your feedback. Can you provide your sharding configuration and create table statement?

**Comment by pcyan** [comment]
> Hi @pcyan, thank you for your feedback. Can you provide your sharding configuration and create table statement?

@strongduanmu exception occurred while parsing sql, not about the configuration or create table statement。just use the case below and you will get ClassCastException
```
def pagination_sql4 = """
    select * from (
    select t1."num",rownum as "rownum_" from(
        select 'a'  as "num" from dual)t1)
    where t2."rownum_" > ? and t2."rownum_" <= (?+?)
"""
```

seems that shardingSphere cannot deal with common expression while pagination

**Comment by xinglijun1973** [comment]
@pcyan  It seems that the table alias `t2` is missing from the above statement. I use the following statement to execute correctly: 
`select *
  from (select t1."num", rownum as "rownum_"
          from (select 'a' as "num" from dual) t1) t2
 where t2."rownum_" > 0
   and t2."rownum_" <= 10
`

**Comment by pcyan** [comment]
fixed in ShardingSphere 5.2.0
